go/teams/box_audit.go

```diff
@@ -1149,3 +1166,43 @@ func randomKnownTeamID(mctx libkb.MetaContext) (teamID *keybase1.TeamID, err err
 	}
 	return &knownTeamIDs[idx.Int64()], nil
 }
+
+func (a *BoxAuditor) getDelayedSlotForTeam(teamID keybase1.TeamID) bool {
+	a.delayedSlotsMutex.Lock()
+	defer a.delayedSlotsMutex.Unlock()
+	found := a.delayedSlots[teamID]
+	if !found {
+		a.delayedSlots[teamID] = true
+	}
+	return !found
+}
+
+func (a *BoxAuditor) clearDelayedSlotForTeam(teamID keybase1.TeamID) {
+	a.delayedSlotsMutex.Lock()
+	defer a.delayedSlotsMutex.Unlock()
+	delete(a.delayedSlots, teamID)
+}
+
+func (a *BoxAuditor) ScheduleDelayedBoxAuditTeam(mctx libkb.MetaContext, teamID keybase1.TeamID) {
+	defer mctx.Trace(fmt.Sprintf("BoxAuditor#ScheduleDelayedBoxAuditTeam(%s)", teamID), func() error { return nil })()
+
+	if !a.getDelayedSlotForTeam(teamID) {
+		mctx.Debug("not scheduling delayed audit, since one is already in progress")
```

Comment:
only one team id can be in the queue at a time, but I agree this is needed to prevent several box audits for the same team happening.